### ISSUE TITLE
Additional metrics support

### ISSUE THREAD

**Issue by drusmanbashir** [seung-lab/connected-components-3d#108]
Hi,
I am benefiting from this amazing library in my deep learning research, namely detecting and measuring liver tumours in CT scans. It would be great if we could have a features allowing unitless longest dimension, surface area metrics to  be computed as well. Getting volume is easy since the functions already return voxel counts for labels.

**Comment by william-silversmith** [comment]
Hi!

I'm glad this library is helpful! Total surface area is a good idea. You can get that by manipulating the output of contacts, but a dedicated function would be ideal. Longest dimension is a bit out of scope, but I think you can get it from performing a principal components analysis for example using scikit-learn:

https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html

Will

**Comment by william-silversmith** [comment]
Closing due to lack of activity. Please re-open if you'd like more info.
